### PR TITLE
allow request for new key to confirm entered pass

### PR DESCRIPTION
**Pull request by Aderemi-Adesada**
allow request for new key to prompt user to confirm the password they entered

### CODE REVIEW

**Comment by istudyatuni** [comment]
Maybe move these closures to separate function?

**Comment by Aderemi-Adesada** [comment]
What is the best file to add the function so that both key.rs and init.rs can access

**Comment by aawsome** [comment]
First, this PR IMO doesn't allow to confirm the password, but enforces it.
I personally wouldn't need that, but if you think this is a worthwhile addition, I'm open for it ;-)

About the place: Please not in `rustic_core`. This crate provides core functionality to access repositories and perform operations on it. All UI topics should stay outside - and this is for sure just a UI topic!

**Comment by Aderemi-Adesada** [comment]
@istudyatuni @aawsome  what do you think about having a function called prompt_password_twice in rustic/src/helpers that both init.rs and key.rs can call

**Comment by istudyatuni** [comment]
I'm afraid to make those decisions in this project 😁

**Comment by aawsome** [comment]
As discussed @Aderemi-Adesada on discord, I also think that the best idea is to replace `rpassword`  with `dialoguer` which already implements this functionality.

**Comment by Aderemi-Adesada** [comment]
rpassword has been replaced with dialoguer in init.rs and key.rs when prompting for new key

File: src/commands/init.rs [hunk 1]
```diff
@@ -80,12 +80,19 @@ pub(crate) fn save_config(
...
+                .interact();
+
+            let password = match password_result {
+                Ok(it) => it,
+                Err(err) => {
+                    status_err!("{}", err);
+                    RUSTIC_APP.shutdown(Shutdown::Crash);
+                }
+            };
+            password
```
Comment (aawsome) [inline src/commands/init.rs:95]: No need for new variables which are immediately used in the next command. You can use `match Password::new() ... { ... }` here.

File: src/commands/init.rs [hunk 2]
```diff
@@ -80,12 +80,19 @@ pub(crate) fn save_config(
...
     let pass = password.map_or_else(
-        || match prompt_password("enter password for new key: ") {
-            Ok(it) => it,
-            Err(err) => {
-                status_err!("{}", err);
-                RUSTIC_APP.shutdown(Shutdown::Crash);
-            }
+        || -> String {
+            let password_result = Password::new().with_prompt("enter password for new key")
+                .with_confirmation("confirm password", "passwords do not match")
```
Comment (aawsome) [inline src/commands/init.rs:85]: According to the documentation you have to enable empty passwords explicitly. These are currently supported by rustic, so please add `allow_empty_password(true)`.

**Review by aawsome — CHANGES_REQUESTED** [review]
Beside my remarks, I was wondering if we can remove the dependency on `rpassword` completely. Then I remembered some newline handling which needed to be dealt with.
Checking the code I realized that that fix already removed `rpassword` when reading from a file - but obviously I have overseen this in `src/command/key.rs`. Can you please have a look at `read_password_from_reader` in `rates/rustic_core/src/repository.rs` and use this in  `src/command/key.rs` - this should allow to remove `rpassword` completely.

Moreover, can you please add an entry in `changelog/new.txt` and after your changes squash all commits into 1 commit? Thanks a lot!

**Comment by Aderemi-Adesada** [comment]
rpassword have been removed completely and `src/command/key.rs` now uses `read_password_from_reader` as used in `crates/rustic_core/src/repository.rs`

**Comment by aawsome** [comment]
@Aderemi-Adesada there are syntax errors in `lib.rs` - can you have a look at it?

**Comment by Aderemi-Adesada** [comment]
> syntax error

I will fix that right now

**Comment by Aderemi-Adesada** [comment]
syntax error have been fixed

**Comment by aawsome** [comment]
Can you please rebase to main and squash all commits instead of re-merging master back into your branch? That will keep commit history clean. Thanks a lot!

**Comment by aawsome** [comment]
Oh, clicked wrong - actually wanted to run the tests.
Anyway, we'll see in the CI. Thanks a lot @Aderemi-Adesada - looks good to me!

**Comment by Aderemi-Adesada** [comment]
You are welcome

**Pull request merged**